core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -312,25 +312,30 @@ public int hashCode() {
         return nodeId.hashCode();
     }
 
+    private String toString;
```

Comment:
@bleskes Also, an argument can be made for just computing this in the constructor anyway, then making it final because we basically always use `DiscoveryNode#toString` anyway.